lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1095,18 +1123,30 @@ def deploy_vm(self):
         self.configure_disks(vm_obj=vm_obj)
         self.configure_network(vm_obj=vm_obj)
 
-        if len(self.params['customization']) > 0 or len(self.params['networks']) > 0:
+        # Find if we need network customizations (find keys in dictionary that requires customizations)
+        network_changes = False
+        for nw in self.params['networks']:
+            for key in nw:
+                # We don't need customizations for these keys
+                if key not in ('name', 'vlan', 'device_type'):
+                    network_changes = True
+                    break
+
+        if len(self.params['customization']) > 0 or network_changes is True:
             self.customize_vm(vm_obj=vm_obj)
 
         try:
-            if self.should_deploy_from_template():
+            if self.params['template']:
                 # create the relocation spec
                 relospec = vim.vm.RelocateSpec()
-                # Only provide specific host when using ESXi host directly
+
+                # Only select specific host when ESXi hostname is provided
                 if self.params['esxi_hostname']:
                     relospec.host = hostsystem
                 relospec.datastore = datastore
-                relospec.pool = resource_pool
+
+                if self.params['resource_pool']:
+                    relospec.pool = resource_pool
```

Comment:
That is weird, because the check `if self.params['resource_pool']` should return None, which is considered False by python. We do this because we would like an empty string to also fail. So we don't set `relospec.pool`. That's the intent. Can you try to disable both lines and see if it would work then ? Maybe VMware's error messages are playing up on us again (they really are not very helpful in a lot of cases).